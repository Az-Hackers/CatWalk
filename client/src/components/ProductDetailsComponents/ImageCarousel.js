/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/button-has-type */
/* eslint-disable no-unused-vars */
/* eslint-disable react/no-did-update-set-state */
/* eslint-disable max-len */
/* eslint-disable array-callback-return */
/* eslint-disable consistent-return */
/* eslint-disable react/no-array-index-key */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
import React from 'react';
import carouselCSS from './carouselCSS.css';
import noImage from './assets/no-image.png';
import LoadingDiv from './LoadingDiv';
import ThumbnailGallery from './ThumbnailGallery';
import expand from './assets/expand.png';
import ZoomedImage from './ZoomedImage';

class ImageCarousel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentImgIndex: 0,
      fullSizeImages: [],
      thumbnails: [],
      expandedView: false,
      maxWidth: '600px',
      zoomView: false
    };
    this.changeImageIndex = this.changeImageIndex.bind(this);
    this.next = this.next.bind(this);
    this.prev = this.prev.bind(this);
    this.handleExpandedViewChange = this.handleExpandedViewChange.bind(this);
    this.toggleZoomView = this.toggleZoomView.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.style !== this.props.style) {
      this.extractImages();
    }
    if (prevProps.product !== this.props.product) {
      this.setState({ currentImgIndex: 0 });
    }
  }

  handleExpandedViewChange() {
    if (this.state.expandedView) this.setState({ maxWidth: '600px', expandedView: false });
    else this.setState({ maxWidth: '1200px', expandedView: true });
  }

  toggleZoomView() {
    if (this.state.zoomView) this.setState({ zoomView: false });
    else this.setState({ zoomView: true });
  }

  next() {
    if (this.state.currentImgIndex < this.state.fullSizeImages.length - 1) {
      this.changeImageIndex(this.state.currentImgIndex + 1);
    }
  }

  prev() {
    if (this.state.currentImgIndex > 0) {
      this.changeImageIndex(this.state.currentImgIndex - 1);
    }
  }

  changeImageIndex(i) {
    this.setState({ currentImgIndex: i });
  }

  extractImages() {
    const fullSizeImageUrls = [];
    const thumbnailImageUrls = [];
    this.props.style.photos.forEach((image) => {
      fullSizeImageUrls.push(image.url);
      thumbnailImageUrls.push(image.thumbnail_url);
    });

    this.setState({
      fullSizeImages: fullSizeImageUrls,
      thumbnails: thumbnailImageUrls
    });
  }

  render() {
    return (
      <>
        {this.state.zoomView
          ? (
            <ZoomedImage image={this.state.fullSizeImages[this.state.currentImgIndex]} toggleZoomView={this.toggleZoomView} />
          )
          : (
            <div
              className="carousel-container"
              style={{
                maxWidth: this.state.maxWidth
              }}
            >
              {this.props.isFetching ? <LoadingDiv />
                : (
                  <div className="carousel-wrapper">

                    {this.state.currentImgIndex > 0 && <button className="left-arrow" id="carousel-left-arrow" onClick={this.prev}>&lt;</button>}
                    <div className="carousel-content-wrapper">

                      {this.state.thumbnails.length > 1 && <ThumbnailGallery thumbnails={this.state.thumbnails} currentImgIndex={this.state.currentImgIndex} changeImageIndex={this.changeImageIndex} />}
                      <button className="expanded-view" onClick={this.handleExpandedViewChange}>
                        <img src={expand} alt="" style={{ height: '20px', width: '20px' }} />
                      </button>
                      <div className="carousel-content" style={{ transform: `translateX(-${this.state.currentImgIndex * 100}%)` }}>
                        {this.state.fullSizeImages.map((imageUrl, i) => (<img src={imageUrl || noImage} key={i} alt="" onClick={this.state.expandedView ? this.toggleZoomView : null} style={this.state.expandedView ? { cursor: 'zoom-in' } : { cursor: 'default' }} />))}
                      </div>
                    </div>
                    {this.state.currentImgIndex < this.state.fullSizeImages.length - 1 && <button className="right-arrow" id="carousel-right-arrow" onClick={this.next}>&gt;</button>}

                  </div>
                )}
            </div>
          )}

      </>
    );
  }
}
export default ImageCarousel;
