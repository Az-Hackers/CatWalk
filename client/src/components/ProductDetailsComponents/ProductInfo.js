/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
import React from 'react';

const ProductInfo = (props) => (
  <div id="product-info">
    <h1 id="product-name">{props.name}</h1>
    <span id="product-category">
      {props.category}
    </span>
    <br />
    <br />
    {props.style.sale_price ? (
      <React.Fragment>
        <span id="sale-price" className="large">
          $
          {props.style.sale_price}
        </span>
        <br />
        <span id="old-price" className="small">
          $
          {props.style.original_price}
        </span>
      </React.Fragment>
    ) : (
      <span id="original-price" className="large">
        $
        {props.style.original_price}
      </span>
    )}
    <br />
  </div>
);

export default ProductInfo;
