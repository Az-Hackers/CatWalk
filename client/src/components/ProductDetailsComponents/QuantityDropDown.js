/* eslint-disable no-plusplus */
/* eslint-disable max-len */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
import React from 'react';

class QuantityDropDown extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  populateDropDown() {
    const array = [];
    if (this.props.item === undefined) return [];
    for (let x = 1; x <= this.props.item.quantity && x <= 15; x++) {
      array.push(x);
    }
    return array;
  }

  render() {
    return (
      <select id="quantity-drop-down" className="medium add-to-cart-input"name="quantity" value={this.props.quantity} onChange={this.props.handleChange}>
        {this.populateDropDown().length === 0 ? <option default>---</option> : <option default> Select Quantity </option> }
        {this.populateDropDown().map((x) => (<option value={x} key={x}>{x}</option>))}
      </select>
    );
  }
}

export default QuantityDropDown;
