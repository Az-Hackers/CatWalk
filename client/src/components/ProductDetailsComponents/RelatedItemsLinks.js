import React from 'react';
import {
  Link,
  withRouter
} from 'react-router-dom';

// eslint-disable-next-line import/no-unresolved
// eslint-disable-next-line import/no-absolute-path
class RelatedItemsLinks extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    return (
      <div id="related-items">
        <span className="large">Related Items</span>
        {this.props.related.length > 0 && (
        <ul>
          {this.props.related.map((id) => (
            <li key={id} className="related-item-link">
              <Link to={`/${id}`} onClick={() => this.props.updateData(id)}>{id}</Link>
            </li>
          ))}
        </ul>
        )}
      </div>
    );
  }
}

export default withRouter(RelatedItemsLinks);
