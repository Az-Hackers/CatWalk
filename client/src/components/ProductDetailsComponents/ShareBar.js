import React from 'react';
import {
  FacebookShareButton,
  PinterestShareButton,
  TwitterShareButton,
  TwitterIcon,
  FacebookIcon,
  PinterestIcon
} from 'react-share';

const ShareBar = ({ id, image }) => {
  const shareUrl = `http://localhost:3000/#/${id}`;
  const message = 'Check out this product!';
  const imageUrl = image;

  return (
    <div id="share-bar">
      <FacebookShareButton
        url={shareUrl}
        quote={message}
        style={{margin: 5}}
      >
        <FacebookIcon size={32} round />
      </FacebookShareButton>

      <TwitterShareButton
        url={shareUrl}
        title={message}
        style={{margin: 5}}
      >
        <TwitterIcon size={32} round />
      </TwitterShareButton>

      <PinterestShareButton
        url={shareUrl}
        media={imageUrl}
        style={{margin: 5}}
      >
        <PinterestIcon size={32} round />
      </PinterestShareButton>
    </div>
  );
};

export default ShareBar;
