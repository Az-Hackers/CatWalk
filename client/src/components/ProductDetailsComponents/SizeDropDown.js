/* eslint-disable array-callback-return */
/* eslint-disable max-len */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
import React from 'react';

class SizeDropDown extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  checkForOutOfStock() {
    let total = 0;
    Object.values(this.props.skus).map(({ quantity }) => {
      total += quantity;
    });
    return total > 0;
  }

  render() {
    return (
      <select id="size-drop-down" className="medium add-to-cart-input" name="sku" value={this.props.sku} onChange={this.props.handleChange} style={this.props.sizeError ? { borderColor: 'red' } : { borderColor: 'black' }}>
        {this.checkForOutOfStock() ? <option default>Select Size</option> : <option default>OUT OF STOCK</option>}
        {Object.keys(this.props.skus).map((sku) => (
          this.props.skus[sku].quantity > 0 && <option value={sku} key={sku} name="sku">{this.props.skus[sku].size}</option>
        ))}
      </select>
    );
  }
}

export default SizeDropDown;
