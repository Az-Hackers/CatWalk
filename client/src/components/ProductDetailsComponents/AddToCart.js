/* eslint-disable array-callback-return */
/* eslint-disable no-console */
/* eslint-disable react/no-unused-state */
/* eslint-disable react/no-did-update-set-state */
/* eslint-disable max-len */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React from 'react';
import QuantityDropDown from './QuantityDropDown';
import SizeDropDown from './SizeDropDown';

class AddToCart extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      sku: '',
      quantity: 0,
      inOutfit: false,
      quantityError: false,
      sizeError: false
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.checkForOutOfStock = this.checkForOutOfStock.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (prevProps !== this.props) {
      this.setState({
        sku: '',
        quantity: 0
      });
    }
  }

  handleSubmit(e) {
    e.preventDefault();
    if (this.state.sku === 'Select Size' || !this.state.sku) this.setState({ sizeError: true });
    else if (this.state.quantity === 'Select Quantity' || !this.state.quantity) this.setState({ quantityError: true });
    else {
      console.log(this.state.quantity, this.props.style.skus[this.state.sku].size, this.props.style.name, 'Added to Cart!');
      this.setState({ sizeError: false, quantityError: false });
    }
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
    this.setState({ sizeError: false });
    this.setState({ quantityError: false });
  }

  checkForOutOfStock() {
    let total = 0;
    Object.values(this.props.style.skus).map(({ quantity }) => {
      total += quantity;
    });
    return total > 0;
  }

  render() {
    return (
      <div id="add-to-cart">
        {this.props.isFetching || (
          <form id="add-to-cart-form" onSubmit={this.handleSubmit}>
            <SizeDropDown skus={this.props.style.skus} handleChange={this.handleChange} sku={this.state.sku} sizeError={this.state.sizeError} checkForOutOfStock={this.checkForOutOfStock} />
            <QuantityDropDown item={this.props.style.skus[this.state.sku]} handleChange={this.handleChange} quantity={this.state.quantity} quantityError={this.state.quantityError} />
            {this.checkForOutOfStock() && <input type="submit" value="Add to Cart!" id="add-to-cart-button" className="medium add-to-cart-input" />}
            {this.state.sizeError && <span className="add-to-cart-error">Please Select a Size!</span>}
            {this.state.quantityError && <span className="add-to-cart-error">Please Select a Quantity!</span>}
          </form>
        )}
      </div>
    );
  }
}
export default AddToCart;
