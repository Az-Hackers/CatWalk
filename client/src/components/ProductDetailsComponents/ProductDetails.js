/* eslint-disable import/no-named-as-default-member */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
import React from 'react';
import ProductInfo from './ProductInfo';
import StyleSelector from './StyleSelector';
import ImageGallery from './ImageGallery';
import AddToCart from './AddToCart';
import ShareBar from './ShareBar';

class ProductDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentStyle: {
        skus: {
          '': { quantity: 0, size: ' ' }
        },
        photos: [{ url: '' }]
      }
    };
    this.changeStyle = this.changeStyle.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.styles !== this.props.styles) {
      this.findDefaultStyle();
    }
  }

  changeStyle(style) {
    this.setState({ currentStyle: style });
  }

  findDefaultStyle() {
    const stylesArray = this.props.styles.results || [];
    stylesArray.forEach((style) => {
      if (style['default?']) this.changeStyle(style);
    });
  }

  render() {
    return (
      <div id="product-details">
        <h1>Product Details</h1>
        <ImageGallery style={this.state.currentStyle} product={this.props.id} />
        <ProductInfo
          name={this.props.productData.name}
          style={this.state.currentStyle}
          category={this.props.productData.category}
        />
        <span id="product-slogan">
          Slogan:
          {this.props.productData.slogan}
        </span>
        <br />
        <span id="product-desc">
          Description:
          {this.props.productData.description}
        </span>
        <br />
        <StyleSelector
          styles={this.props.styles || []}
          currentStyle={this.state.currentStyle}
          changeStyle={this.changeStyle}
        />
        <AddToCart style={this.state.currentStyle} />
        <ShareBar id={this.props.id} image={this.state.currentStyle.photos[0].url} />
      </div>

    );
  }
}

export default ProductDetails;
