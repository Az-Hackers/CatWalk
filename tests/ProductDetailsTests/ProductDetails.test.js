import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ProductInfo from '../../client/src/components/ProductDetailsComponents/ProductInfo.js';
import StyleSelector from '../../client/src/components/ProductDetailsComponents/StyleSelector.js';
import AddToCart from '../../client/src/components/ProductDetailsComponents/AddToCart.js';
import ShareBar from '../../client/src/components/ProductDetailsComponents/ShareBar.js';
import ImageCarousel from '../../client/src/components/ProductDetailsComponents/ImageCarousel.js';
import ThumbnailGallery from '../../client/src/components/ProductDetailsComponents/ThumbnailGallery.js';
import ZoomedImage from '../../client/src/components/ProductDetailsComponents/ZoomedImage.js';
import ProductDetails from '../../client/src/components/ProductDetailsComponents/ProductDetails.js';
import QuantityDropDown from '../../client/src/components/ProductDetailsComponents/QuantityDropDown.js';
import SizeDropDown from '../../client/src/components/ProductDetailsComponents/SizeDropDown.js';
import RelatedItemsLinks from '../../client/src/components/ProductDetailsComponents/RelatedItemsLinks.js';

// shallow render tests for Product Details components

let props = {
  style: {
    skus: { '000': { quantity: 1, size: 'X' } },
    name: 'XXXX',
    'default?': true,
    photos: [{ thumbnail_url: '', url: '' }],
    original_price: '0.00'
  },
  productId: '00000',
  productData: {
    category: 'category',
    default_price: '0.00',
    descirption: 'description',
    id: 0,
    name: 'XXXX',
    slogan: 'slogan',
    styles: {
      productId: '00000',
      results: [{}]
    }
  },
  category: 'category',
  name: 'name',
  styles: {
    productId: '00000',
    results: [{ photos: [{ thumbnail_url: '' }], style_id: '00001' }]
  },
  image: '',
  fn: () => null,
  sku: {quantity: 1, size: 'X'}
};

describe('Product Details components should all successfully render in isolation', () => {
  it('should render parent ProductDetails component', () => {
    const wrapper = shallow(<ProductDetails productData={props.productData} styles={props.styles} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render AddToCart', () => {
    const wrapper = shallow(<AddToCart style={props.style} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render ImageCarousel', () => {
    const wrapper = shallow(<ImageCarousel product={props.productId} style={props.style} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render ProductInfo', () => {
    const wrapper = shallow(<ProductInfo category={props.category} name={props.name} style={props.style} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render QuantityDropDown', () => {
    const wrapper = shallow(<QuantityDropDown handleChange={props.fn} item={props.sku} quantity={props.sku.quantity} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render RelatedItemsLinks', () => {
    const wrapper = shallow(<RelatedItemsLinks related={[0]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render SizeDropDown', () => {
    const wrapper = shallow(<SizeDropDown handleChange={props.fn} checkForOutOfStock={props.fn} sku={props.name} skus={props.style.skus} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render StyleSelector', () => {
    const wrapper = shallow(<StyleSelector styles={props.styles} currentStyle={props.style} changeStyle={props.fn} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render ThumbnailGallery', () => {
    const wrapper = shallow(<ThumbnailGallery changeImageIndex={props.fn} currentImgIndex={0} thumbnails={[props.image]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render ShareBar', () => {
    const wrapper = shallow(<ShareBar id={props.id} image={props.image} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render ZoomedImage', () => {
    const wrapper = shallow(<ZoomedImage toggleZoomView={props.fn} image={props.image} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
