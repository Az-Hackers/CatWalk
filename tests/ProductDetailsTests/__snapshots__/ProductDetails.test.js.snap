// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Details components should all successfully render in isolation should render AddToCart 1`] = `
<div
  id="add-to-cart"
>
  <form
    id="add-to-cart-form"
    onSubmit={[Function]}
  >
    <SizeDropDown
      checkForOutOfStock={[Function]}
      handleChange={[Function]}
      sizeError={false}
      sku=""
      skus={
        Object {
          "000": Object {
            "quantity": 1,
            "size": "X",
          },
        }
      }
    />
    <QuantityDropDown
      handleChange={[Function]}
      quantity={0}
      quantityError={false}
    />
    <input
      className="medium add-to-cart-input"
      id="add-to-cart-button"
      type="submit"
      value="Add to Cart!"
    />
  </form>
</div>
`;

exports[`Product Details components should all successfully render in isolation should render ImageCarousel 1`] = `
<Fragment>
  <div
    className="carousel-container"
    style={
      Object {
        "maxWidth": "600px",
      }
    }
  >
    <div
      className="carousel-wrapper"
    >
      <div
        className="carousel-content-wrapper"
      >
        <button
          className="expanded-view"
          onClick={[Function]}
        >
          <img
            alt=""
            src="test-file-stub"
            style={
              Object {
                "height": "20px",
                "width": "20px",
              }
            }
          />
        </button>
        <div
          className="carousel-content"
          style={
            Object {
              "transform": "translateX(-0%)",
            }
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Product Details components should all successfully render in isolation should render ProductInfo 1`] = `
<div
  id="product-info"
>
  <h1
    id="product-name"
  >
    name
  </h1>
  <span
    id="product-category"
  >
    category
  </span>
  <br />
  <br />
  <span
    className="large"
    id="original-price"
  >
    $
    0.00
  </span>
  <br />
</div>
`;

exports[`Product Details components should all successfully render in isolation should render QuantityDropDown 1`] = `
<select
  className="medium add-to-cart-input"
  id="quantity-drop-down"
  name="quantity"
  onChange={[Function]}
  style={
    Object {
      "borderColor": "black",
    }
  }
  value={1}
>
  <option
    default={true}
  >
     Select Quantity 
  </option>
  <option
    key="1"
    value={1}
  >
    1
  </option>
</select>
`;

exports[`Product Details components should all successfully render in isolation should render RelatedItemsLinks 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Product Details components should all successfully render in isolation should render ShareBar 1`] = `
<div
  id="share-bar"
>
  <ForwardRef(ShareButton-facebook)
    id="facebook-share-button"
    quote="Check out this product!"
    style={
      Object {
        "margin": 5,
      }
    }
    url="http://localhost:3000/#/undefined"
  >
    <Icon
      bgStyle={Object {}}
      borderRadius={0}
      iconFillColor="white"
      id="facebook-icon"
      round={true}
      size={32}
    />
  </ForwardRef(ShareButton-facebook)>
  <ForwardRef(ShareButton-twitter)
    id="twitter-share-button"
    style={
      Object {
        "margin": 5,
      }
    }
    title="Check out this product!"
    url="http://localhost:3000/#/undefined"
  >
    <Icon
      bgStyle={Object {}}
      borderRadius={0}
      iconFillColor="white"
      id="twitter-icon"
      round={true}
      size={32}
    />
  </ForwardRef(ShareButton-twitter)>
  <ForwardRef(ShareButton-pinterest)
    id="pinterest-share-button"
    media=""
    style={
      Object {
        "margin": 5,
      }
    }
    url="http://localhost:3000/#/undefined"
  >
    <Icon
      bgStyle={Object {}}
      borderRadius={0}
      iconFillColor="white"
      id="pinterest-icon"
      round={true}
      size={32}
    />
  </ForwardRef(ShareButton-pinterest)>
</div>
`;

exports[`Product Details components should all successfully render in isolation should render SizeDropDown 1`] = `
<select
  className="medium add-to-cart-input"
  id="size-drop-down"
  name="sku"
  onChange={[Function]}
  style={
    Object {
      "borderColor": "black",
    }
  }
  value="name"
>
  <option
    default={true}
  >
    OUT OF STOCK
  </option>
  <option
    key="000"
    name="sku"
    value="000"
  >
    X
  </option>
</select>
`;

exports[`Product Details components should all successfully render in isolation should render StyleSelector 1`] = `
<div
  id="style-selector"
>
  <span
    className="large"
    id="style-name"
  >
    XXXX
  </span>
  <br />
  <div
    id="style-selector-list"
  >
    <div
      className="style-selector-item"
      key="00001"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(test-file-stub)",
        }
      }
    />
  </div>
</div>
`;

exports[`Product Details components should all successfully render in isolation should render ThumbnailGallery 1`] = `
<div
  id="thumbnail-gallery-container"
>
  <div
    id="thumbnail-gallery-content-container"
  >
    <div
      id="thumbnail-gallery"
      style={
        Object {
          "transform": "translateY(0px)",
        }
      }
    >
      <img
        alt=""
        className="thumbnail-gallery-item selected-thumbnail"
        key="0"
        name={0}
        onClick={[Function]}
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;

exports[`Product Details components should all successfully render in isolation should render ZoomedImage 1`] = `
<div
  className="image-modal"
>
  <div
    className="zoomed-image-container"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "cursor": "zoom-out",
      }
    }
  >
    <img
      className="zoomed-image"
      onMouseMove={[Function]}
      src=""
      style={
        Object {
          "transform": "translateX(100%) translateY(100%) scale(2)",
        }
      }
    />
  </div>
</div>
`;

exports[`Product Details components should all successfully render in isolation should render parent ProductDetails component 1`] = `
<div>
  <ProductDetails
    onClick={[Function]}
    productData={
      Object {
        "category": "category",
        "default_price": "0.00",
        "descirption": "description",
        "id": 0,
        "name": "XXXX",
        "slogan": "slogan",
        "styles": Object {
          "productId": "00000",
          "results": Array [
            Object {},
          ],
        },
      }
    }
    styles={
      Object {
        "productId": "00000",
        "results": Array [
          Object {
            "photos": Array [
              Object {
                "thumbnail_url": "",
              },
            ],
            "style_id": "00001",
          },
        ],
      }
    }
  />
</div>
`;
